torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -695,10 +695,11 @@ def _get_grad_inner_tensor(self, grad: torch.Tensor) -> torch.Tensor:
             if isinstance(grad, AsyncCollectiveTensor):
                 grad = grad.wait()
             assert isinstance(grad, DTensor), f"{type(grad)}"
-            if any(pl.is_partial() for pl in grad.placements):
```

Comment:
Previously, we changed any partial placements to replicate mainly targeting the case where replicated `RMSNorm.weight` had partial gradients, and we needed to trigger the all-reduce by converting from partial to replicated.

However, for the `pos_embeddings` in our toy `Transformer` class, we have `Shard(0)` placement and still `Partial` grad. We _do not_ want to convert it to `Replicate` but rather to `Shard(0)`.

Reply:
I think that size the viewing into reduce-scatter output uses `torch.as_strided`, we were silently handling a larger/replicated gradient for `pos_embeddings.weight`.